poly/src/polynomial/univariate/dense.rs

```diff
@@ -262,6 +261,37 @@ impl<'a, 'b, F: Field> Add<&'a DensePolynomial<F>> for &'b DensePolynomial<F> {
     }
 }
 
+impl<'a, 'b, F: Field> Add<&'a SparsePolynomial<F>> for &'b DensePolynomial<F> {
+    type Output = DensePolynomial<F>;
+
+    #[inline]
+    fn add(self, other: &'a SparsePolynomial<F>) -> DensePolynomial<F> {
+        let result = if self.is_zero() {
+            other.clone().into()
+        } else if other.is_zero() {
+            self.clone()
+        } else {
+            let mut result = self.clone();
+            // If `other` has higher degree than `self`, create a dense vector
+            // storing the upper coefficients of the addition
+            let mut upper_coeffs = match other.degree() > result.degree() {
+                true => vec![F::zero(); other.degree() - result.degree()],
+                false => Vec::new(),
+            };
+            for (pow, coeff) in other.iter() {
+                if *pow <= result.degree() {
+                    result.coeffs[*pow] += coeff;
+                } else {
+                    upper_coeffs[*pow - result.degree() - 1] = *coeff;
+                }
+            }
+            result.coeffs.extend(upper_coeffs);
+            result
+        };
+        result
+    }
+}
```

Comment:
Probably in this case we should add `AddAssign`/`SubAssign` impls too.

Reply:
added, also with tests.